collectives/routes/event.py

```diff
@@ -819,7 +819,7 @@ def register_user(event_id):
                     "inscription ne sera confirmée qu'après renouvellement"
                 )
 
-            if payment_required and registration.status != RegistrationStatus.Active:
+            if payment_required and registration.status not in RegistrationStatus.valid:
```

Comment:
Je vois pas où `RegistrationStatus.valid` est défini, ça ne devrait pas `not registration.is_valid()` plutôt ?

Reply:
Hum... J'ai un gros doute. Avant, la condition était:

>  if payment_required and registration.status != RegistrationStatus.Active:

Seulement, je pense que `registration.status` était tout le temps à `None`... En plus, la condition sert à set le `registration.status`. J'ai l'impression que ca marchait par hasard, non?
Est-ce qu'on ne devrais pas juste garder `if payment_required:` ?

Reply:
En gros pour l'instant les encadrants peuvent "ré-inscrire" un participant en remettant son nom dans l'auto-complete, dans ce cas le registration.status ne sera pas nul. Ca servait par exemple à ré-ajouter des personnes refusées (par erreur ou non) avant qu'on ait la table des inscrits en bas de page. Mais du coup si quelqu'un a déjà payé, il ne faut pas le repasser en PaymentPending, on laisse le status à Active/Présent.
Cela dit maintenant qu'on a la table des inscrits pour gérer les status il serait peut être plus simple de ne pas autoriser la réinscription d'un participant existant